betterdiscordctl

```diff
@@ -373,49 +285,31 @@ bdc_upgrade() {
 # Implementation functions
 
 bdc_main() {
-  if [[ -z $discord ]]; then
+  if [[ -z $modules ]]; then
     if [[ $snap ]]; then bdc_snap
     elif [[ $flatpak ]]; then bdc_flatpak
-    elif [[ $nix ]]; then bdc_nix
     else bdc_scan; fi
   else
-    flavor=$flavors
-    # --discord and --modules
-    [[ -z $modules ]] && die_with_help 'ERROR: "--discord" requires "--modules" to also be set.'
-    [[ -d $discord ]] || die 'ERROR: Discord installation not found.'
     [[ -d $modules ]] || die 'ERROR: Discord modules directory not found.'
+    : "${modules%/*/modules}"; flavor=${_##*/discord}
   fi
-  [[ -d $discord ]] || die 'ERROR: Discord installation not found. Try specifying it with "--discord".'
   core=$modules/discord_desktop_core
-  [[ -d $core ]] || die "ERROR: Directory 'discord_desktop_core' not found in $(readlink -f "$modules")"
+  [[ -d $core ]] || die "ERROR: Directory 'discord_desktop_core' not found in $modules"
 }
 
 bdc_scan() {
-  for scandir in "${scan[@]}"; do
-    verbose 2 "VV: Scanning $scandir"
-    for flavor in "${flavors[@]}"; do
-      verbose 2 "VV: Trying flavor '$flavor'"
-      shopt -s nocaseglob
-      for discord in "$scandir"/discord?(-)"$flavor"; do
-        shopt -u nocaseglob
-        if [[ -d $discord ]]; then
-          verbose 1 "V: Using Discord at $discord"
-          discord_config=$xdg_config/discord${flavor,,}
-          if [[ ! -d $discord_config ]]; then
-            printf 'WARN: Config directory not found for Discord %s (%s, %s).\n' \
-              "$flavor" "$discord" "$discord_config" >&2
-            continue 2
-          fi
-          if [[ -z $modules ]]; then
-            bdc_find_modules
-          else
-            # --modules
-            [[ -d $modules ]] || die 'ERROR: Discord modules directory not found.'
-          fi
-          break 3
-        fi
-      done
-    done
+  for flavor in "${flavors[@]-}"; do
```

Comment:
What does this hyphen do?

Reply:
`flavors` should always be defined. Did you run into a situation when it wasn't?

Reply:
Using `-f ''` to install to stable does not work and requires `-f ':'`. This change fixes that.